core/utils.h

```diff
@@ -1946,6 +1946,14 @@ notify(syslog_event_type_t priority, bool internal, bool synch,
 #define SYSLOG_INTERNAL_CRITICAL_ONCE(...) \
     DODEBUG_ONCE(SYSLOG_INTERNAL_CRITICAL(__VA_ARGS__))
 
+#define MSG_COULDNT_SYNCH_WITH_ALL_THREADS_CODE 42
+
+#define fatal_error(id, sub, ...) \
+    do { \
+      SYSLOG_COMMON(false, SYSLOG_CRITICAL, id, sub, __VA_ARGS__); \
+      exit_process_syscall(MSG_##id##_CODE); \
```

Comment:
This is unix-only -- use os_terminate_with_code

Reply:
That might explain the windows error!